File: common/src/main/java/com/jmsoftware/maf/common/domain/DeletedField.java
package com.jmsoftware.maf.common.domain;

import lombok.Getter;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;

/**
 * Description: DeletedField, change description here.
 *
 * @author 钟俊 （zhongjun）, email: zhongjun@toguide.cn, date: 1/13/2021 6:32 PM
 */
@Slf4j
@Getter
@ToString
public enum DeletedField {
    /**
     * Not deleted
     */
    NOT_DELETED("N", "not deleted"),
    /**
     * Deleted
     */
    DELETED("Y", "deleted");

    /**
     * The Value.
     */
    private final String value;
    /**
     * The Description.
     */
    private final String description;

    /**
     * Instantiates a new Delete flag.
     *
     * @param value       the value
     * @param description the description
     */
    DeletedField(String value, String description) {
        this.value = value;
        this.description = description;
    }
}

File: common/src/main/java/com/jmsoftware/maf/common/domain/DeletedField.kt
package com.jmsoftware.maf.common.domain

/**
 * # DeletedField
 *
 * Description: DeletedField, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 8:17 PM
 */
enum class DeletedField(
    /**
     * The Value.
     */
    val value: String,
    /**
     * The Description.
     */
    val description: String
) {
    /**
     * Not deleted
     */
    NOT_DELETED("N", "not deleted"),

    /**
     * Deleted
     */
    DELETED("Y", "deleted");
}
